### PR TITLE
Initial RHEL 8 CIS profile

### PR DESCRIPTION
**Pull request by shawndwells**
Initial CIS profile.

### CODE REVIEW

**Comment by ggbecker** [comment]
@openscap-ci test this

File: linux_os/guide/system/auditing/auditd_configure_rules/audit_rules_networkconfig_modification/rule.yml [hunk 1]
```diff
@@ -41,7 +41,7 @@ references:
     srg@rhel6: SRG-OS-999999
     nist@rhel6: AC-3(10)
     nist-csf@rhel6: PR.AC-4,PR.AC-6,PR.PT-3
-    cis: 5.2.6
+    cis@rhel8: 4.18
```
Comment (ggbecker) [inline linux_os/guide/system/auditing/auditd_configure_rules/audit_rules_networkconfig_modification/rule.yml]:
```suggestion
    cis@rhel8: "4.18"
```
The build system interprets as a float number in this case. It needs to be double quoted

File: linux_os/guide/system/network/network-ipv6/disabling_ipv6/kernel_module_ipv6_option_disabled/rule.yml [hunk 2]
```diff
@@ -19,9 +19,12 @@ severity: medium
 
 identifiers:
     cce@rhel6: 27153-6
+    cce@rhel7: 82871-5
+    cce@rhel8: 82872-3
 
 references:
     disa@rhel6: "1551"
+    cis@rhel8: 3.6
```
Comment (ggbecker) [inline linux_os/guide/system/network/network-ipv6/disabling_ipv6/kernel_module_ipv6_option_disabled/rule.yml]:
```suggestion
    cis@rhel8: "3.6"
```

The build system interprets as a float number in this case. It needs to be double quoted

File: linux_os/guide/system/network/network-wireless/wireless_software/wireless_disable_interfaces/rule.yml [hunk 3]
```diff
@@ -31,7 +31,7 @@ identifiers:
 references:
     stigid@rhel6: "000293"
     stigid@rhel7: "041010"
-    cis: 4.3.1
+    cis@rhel8: 3.5
```
Comment (ggbecker) [inline linux_os/guide/system/network/network-wireless/wireless_software/wireless_disable_interfaces/rule.yml]:
```suggestion
    cis@rhel8: "3.5"
```
The build system interprets as a float number in this case. It needs to be double quoted

File: linux_os/guide/system/permissions/mounting/kernel_module_usb-storage_disabled/rule.yml [hunk 4]
```diff
@@ -40,6 +40,8 @@ references:
     cobit5: APO13.01,DSS01.04,DSS05.03,DSS05.04,DSS05.05,DSS05.07,DSS05.10,DSS06.03,DSS06.10
     iso27001-2013: A.11.2.6,A.13.1.1,A.13.2.1,A.18.1.4,A.6.2.1,A.6.2.2,A.7.1.1,A.9.2.1,A.9.2.2,A.9.2.3,A.9.2.4,A.9.2.6,A.9.3.1,A.9.4.2,A.9.4.3
     cis-csc: 1,12,15,16,5
+    cis: 8.4,8.5
```
Comment (ggbecker) [inline linux_os/guide/system/permissions/mounting/kernel_module_usb-storage_disabled/rule.yml]:
```suggestion
    cis@rhel7: 8.4,8.5
```
The build system doesn't allow `cis` and `cis@rhel8` at the same time. Are these identifiers specific to `rhel7` instead?

File: rhel8/profiles/cis.profile [hunk 5]
```diff
@@ -0,0 +1,869 @@
...
+    #### 4.1.2.2 Ensure audit logs are not automatically deleted (Scored)
+    - auditd_data_retention_max_log_file_action
+
+    #### 4.1.2.3 Ensure system is disabled when audit logs are full (Scored)
+    ##### space_left_action = email
+    - auditd_data_retention_admin_space_left_action
+    # TO DO: SET TO EMAIL
+
+    ##### action_mail_acct = root
+    - var_auditd_action_mail_acct = root
```
Comment (ggbecker) [inline rhel8/profiles/cis.profile]:
```suggestion
    - var_auditd_action_mail_acct=root
```
The syntax for variables is `var_name=value`. The build system cannot process spaces in this case.

File: linux_os/guide/system/auditing/auditd_configure_rules/audit_rules_networkconfig_modification/rule.yml (hunk 1 above)
Comment (shawndwells) [reply linux_os/guide/system/auditing/auditd_configure_rules/audit_rules_networkconfig_modification/rule.yml]: Ack & fixed.Thansk!

File: linux_os/guide/system/network/network-ipv6/disabling_ipv6/kernel_module_ipv6_option_disabled/rule.yml (hunk 2 above)
Comment (shawndwells) [reply linux_os/guide/system/network/network-ipv6/disabling_ipv6/kernel_module_ipv6_option_disabled/rule.yml]: Ack & fixed. Thanks!

File: linux_os/guide/system/network/network-wireless/wireless_software/wireless_disable_interfaces/rule.yml (hunk 3 above)
Comment (shawndwells) [reply linux_os/guide/system/network/network-wireless/wireless_software/wireless_disable_interfaces/rule.yml]: Ack & fixed

File: linux_os/guide/system/permissions/mounting/kernel_module_usb-storage_disabled/rule.yml (hunk 4 above)
Comment (shawndwells) [reply linux_os/guide/system/permissions/mounting/kernel_module_usb-storage_disabled/rule.yml]: CIS has base mappings, which is what those were... and then they have mappings that are specific to a baseline.

Made up example could be:
CIS 8.1: Use encryption

RHEL 8 CIS 123.1: Use the following ciphers in sshd config

Comment (shawndwells) [reply linux_os/guide/system/permissions/mounting/kernel_module_usb-storage_disabled/rule.yml]: remapped to rhel8

File: rhel8/profiles/cis.profile (hunk 5 above)
Comment (shawndwells) [reply rhel8/profiles/cis.profile]: ack & fixed

File: linux_os/guide/system/permissions/mounting/kernel_module_udf_disabled/rule.yml [hunk 6]
```diff
@@ -8,19 +8,26 @@ description: |-
     {{{ describe_module_disable(module="udf") }}}
     This effectively prevents usage of this uncommon filesystem.
 
+    The <tt>udf</tt> filesystem type is the universal disk format
+    used to implement the ISO/IEC 13346 and ECMA-167 specifications.
+    This is an open vendor filesystem type for data storage on a broad
+    range of media. This filesystem type is neccessary to support
+    writnig DVDs and newer optical disc formats.
```
Comment (redhatrises) [inline linux_os/guide/system/permissions/mounting/kernel_module_udf_disabled/rule.yml]: `s/writnig/writing/g`

File: linux_os/guide/system/permissions/mounting/kernel_module_usb-storage_disabled/rule.yml [hunk 7]
```diff
@@ -40,6 +40,7 @@ references:
     cobit5: APO13.01,DSS01.04,DSS05.03,DSS05.04,DSS05.05,DSS05.07,DSS05.10,DSS06.03,DSS06.10
     iso27001-2013: A.11.2.6,A.13.1.1,A.13.2.1,A.18.1.4,A.6.2.1,A.6.2.2,A.7.1.1,A.9.2.1,A.9.2.2,A.9.2.3,A.9.2.4,A.9.2.6,A.9.3.1,A.9.4.2,A.9.4.3
     cis-csc: 1,12,15,16,5
+    cis@rhel8: "1.1.23"
```
Comment (redhatrises) [inline linux_os/guide/system/permissions/mounting/kernel_module_usb-storage_disabled/rule.yml]: Remove quotes here... greater than 1 decimal point don't need quotes as it is no longer treated as a decimal in yaml.

File: linux_os/guide/system/permissions/mounting/service_autofs_disabled/rule.yml [hunk 8]
```diff
@@ -33,7 +33,8 @@ identifiers:
 references:
     stigid@rhel6: "000526"
     srg@rhel6: SRG-OS-999999
-    cis: 1.1.22
+    cis: 8.4,8.5
+    rhel8cis: 1.1.22
```
Comment (redhatrises) [inline linux_os/guide/system/permissions/mounting/service_autofs_disabled/rule.yml]: Fix your reference here.

File: linux_os/guide/system/permissions/mounting/kernel_module_squashfs_disabled/rule.yml [hunk 9]
```diff
@@ -20,7 +25,7 @@ identifiers:
     cce@ocp4: 82717-0
 
 references:
-    cis: 1.1.1.6
+    rhel8cis: 1.1.1.3
```
Comment (redhatrises) [inline linux_os/guide/system/permissions/mounting/kernel_module_squashfs_disabled/rule.yml]: Fix this reference typo.

File: linux_os/guide/system/permissions/mounting/kernel_module_vfat_disabled/rule.yml [hunk 10]
```diff
@@ -21,7 +26,7 @@ identifiers:
     cce@ocp4: 82720-4
 
 references:
-    cis: 1.1.1.8
+    rhel8cis: 1.1.1.2
```
Comment (redhatrises) [inline linux_os/guide/system/permissions/mounting/kernel_module_vfat_disabled/rule.yml]: Fix this reference typo

Comment (shawndwells) [reply linux_os/guide/system/permissions/mounting/kernel_module_vfat_disabled/rule.yml]: fixed

File: linux_os/guide/system/permissions/mounting/kernel_module_squashfs_disabled/rule.yml (hunk 9 above)
Comment (shawndwells) [reply linux_os/guide/system/permissions/mounting/kernel_module_squashfs_disabled/rule.yml]: fixed

File: linux_os/guide/system/permissions/mounting/service_autofs_disabled/rule.yml (hunk 8 above)
Comment (shawndwells) [reply linux_os/guide/system/permissions/mounting/service_autofs_disabled/rule.yml]: fixed

File: linux_os/guide/system/permissions/mounting/kernel_module_usb-storage_disabled/rule.yml (hunk 7 above)
Comment (shawndwells) [reply linux_os/guide/system/permissions/mounting/kernel_module_usb-storage_disabled/rule.yml]: fixed (note earlier feedback said to add the quotes)

File: linux_os/guide/system/permissions/mounting/kernel_module_udf_disabled/rule.yml (hunk 6 above)
Comment (shawndwells) [reply linux_os/guide/system/permissions/mounting/kernel_module_udf_disabled/rule.yml]: fixed

File: linux_os/guide/system/permissions/mounting/kernel_module_usb-storage_disabled/rule.yml (hunk 7 above)
Comment (shawndwells) [reply linux_os/guide/system/permissions/mounting/kernel_module_usb-storage_disabled/rule.yml]: fixed

**Comment by shawndwells** [comment]
@redhatrises feedback addressed and commits squashed. Awaiting the test harness.

**Pull request merged**